bin/upgrade-groups.bash

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+here="$(dirname "$(readlink -f "$0")")"
+usage() {
+    echo "COMMANDS:" 1>&2
+    echo "  <prefix> list-groups                    List groups " 1>&2
+    echo "      args: [<options>]" 1>&2
+    echo "  <prefix> upgrade-groups                    Run upgrade playbook " 1>&2
+    echo "      args: [--skip-download]" 1>&2
+    exit 1
+}
+
+
+if [ $# -lt 2 ]; then
+    usage
+else
+    export prefix="${1}"
+fi
+
+# shellcheck source=bin/common.bash
+source "${here}/common.bash"
+
+# shellcheck source=bin/common.bash
+source "${here}/inventory-parser.bash"
+
+list_groups() {
+    groups=$(readInventoryGroups ${config[groups_inventory_file]})
+    for group in $groups; do
+        log_info "- $group"
+    done
+}
+
+upgrade_groups() {
+    local -a groups
+    for group in $(readInventoryGroups "${config[groups_inventory_file]}" | sort); do
+      if ! [[ "${group}" =~ ^(all|etcd|kube_node|.*:.*)$ ]]; then
+        groups+=("${group}")
+      fi
+    done
+
+    local -A group_lengths
+    group_lengths["kube_control_plane"]="$(getGroupHosts "${config[groups_inventory_file]}" "kube_control_plane" | wc -w)"
+    for group in "${groups[@]}"; do
+      group_lengths["${group}"]="$(getGroupHosts "${config[groups_inventory_file]}" "${group}" | wc -w)"
+    done
+
+    pushd "${kubespray_path}"
+    skip_tags="multus"
+    if [[ "${1}" == "--skip-download" ]]; then
+      skip_tags+=",download"
+    fi
+    ansible-playbook upgrade-cluster.yml -b -i "${config[groups_inventory_file]}" --skip-tags=${skip_tags} --limit "kube_control_plane[0]"
```

Comment:
Why do we want to upgrade only the first control plane node? Shouldn't this be

```suggestion
    ansible-playbook upgrade-cluster.yml -b -i "${config[groups_inventory_file]}" --skip-tags=${skip_tags} --limit "kube_control_plane"
```

Reply:
This is so it will upgrade the first control-plane node before it runs the rest in parallel with other nodes.